packages/nimble-components/src/theme-provider/index.ts

```diff
@@ -33,19 +54,44 @@ export const theme = DesignToken.create<Theme>({
  * @internal
  */
 export class ThemeProvider extends FoundationElement {
-    @attr({
-        attribute: 'direction'
-    })
+    @attr()
+    public override lang = '';
+
+    @attr()
     public direction: Direction = Direction.ltr;
 
-    @attr({
-        attribute: 'theme'
-    })
+    @attr()
     public theme: Theme = Theme.light;
 
+    public get validity(): ValidityObject {
+        return {
+            invalidLang: this.langIsInvalid
+        };
+    }
+
+    private langIsInvalid = false;
+
+    public langChanged(
+        _prev: string | undefined | null,
+        next: string | undefined | null
+    ): void {
+        this.langIsInvalid = false;
+        if (
+            next !== undefined
+            && next !== null
+            && next !== ''
+            // eslint-disable-next-line no-cond-assign
+            && !(this.langIsInvalid = isInvalidLang(next))
```

Comment:
I think we should attempt not to have to disable our lint rules unless we have a good reason to. Is there a reason you decided that this would be best written in a way that requires disabling a lint rule?

Reply:
I want to avoid calling `isInvalidLang` unless we need to, i.e. when `next` is truthy, and if I do the assignment inside the `if` block, I have to duplicate the `else` with `lang.deleteValueFor`. The documentation for the rule itself says, "There are valid reasons to use assignment operators in conditional statements." For these reasons, I just disabled the rule in this case.

Reply:
> The documentation for the rule itself says, "There are valid reasons to use assignment operators in conditional statements." For these reasons, I just disabled the rule in this case.

Eslint's description of the rule does not necessarily imply anything about the NI JavaScript Styleguide configuration of the rule. We follow airbnb's convention without modification which always disallows without description of exceptions: 

https://github.com/airbnb/javascript/blob/cda44dad403979d313a4b02ce55f0bb6cc3a24b2/packages/eslint-config-airbnb-base/rules/errors.js#L23C1-L24C43

So I'm going to push back and ask to come up with a pattern that avoids the mutation in the conditional